src/frontend/stmclient.cc

```diff
@@ -126,7 +127,7 @@ void STMClient::init( void )
   if ( (escape_key_env = getenv( "MOSH_ESCAPE_KEY" )) != NULL ) {
     if ( strlen( escape_key_env ) == 1 ) {
       escape_key = (int)escape_key_env[0];
-      if ( escape_key > 0 && escape_key < 128 ) {
+      if ( escape_key > 0 || escape_key < 128 ) {
```

Comment:
It should be &&.

Meaning that any single character ASCII value is accepted (except NUL) and if the environment contains something else (e.g. a ISO-8859-1 character, which also could be a start sequence of UTF-8 combined character) it's just silently ignored. Then the magic is to separate ASCII-characters below 32 so that they can be referred as CTRL-something in info strings rather than by the character value itself.

BTW, there is still one small bug in escape character info string.  If the connection is offline (e.g. network connection goes broken) and mosh starts to show the info string about how to kill the process. That doesn't show the correct escape string if it has been modified by the environment but it rather says always CTRL-^. I've been thinking about fixing it sometimes but always forgotten. It is probably a one or two-liner, so please feel free.